scripts.js

```diff
@@ -1 +1,331 @@
-console.log('Hello world');
\ No newline at end of file
+// Exercise 1 - getTheSmallestNumber function
+
+function getSmallestNumber(arr) {
+        const sortedArr = arr.slice().sort((a, b) => a - b);
+        return sortedArr[0];
+}
+
+getSmallestNumber([2, -5, 10, 1, 4]);
+getSmallestNumber([200, 25, 4, 123, 87]);
+
+function getSmallestNumberByLoop (arr) {
+        let smallestNumber = arr[0];
+        for (let i = 0; i < arr.length; i++) {
+                if(arr[i] < smallestNumber ) {
+                        smallestNumber = arr[i]
+                }
+        }
+        return smallestNumber;
+}
+
+getSmallestNumberByLoop([2, -5, 10, 1, 4]);
+getSmallestNumberByLoop([200, 25, 4, 123, 87]);
+
+// Excercise 2 - write the getSquaredNumbers function - don't modify the array passed as an argument
+
+function getSquaredNumbers(arr) {
+        const squaredArray = [];
+        for (let i = 0; i < arr.length; i++) {
+                squaredArray.push(arr[i] * arr[i]);
+        }
+        return squaredArray;
+}
+
+// Excercise 3 - write the getReversedString function
+
+function getReversedString(str) {
+        return str.split("").reverse();
+}
+getReversedString("Hello")
+
+// Excercise 4 - Write the isPalindrome  function.
+// Inside the isPalindrome  function, call the getReversedString  function.
+// It should be case-insensitive and shouldn't take spaces into consideration.
+
+function isPalindrome(str) {
+        function getReversedString(str) {
+                return str.split("").reverse();
+        }
```

Comment:
https://github.com/szymon-czudowski/WANACODE-arrays-and-loops/pull/1#discussion_r1354328231

As pointed before, you really don't need to declare same `getReversedString` function twice